packages/next/client/dev/error-overlay/hot-dev-client.js

```diff
@@ -342,7 +345,11 @@ function tryApplyUpdates(onHotUpdateSuccess) {
 
     if (isUpdateAvailable()) {
       // While we were updating, there was a new update! Do it again.
-      tryApplyUpdates(hasUpdates ? onBuildOk : onHotUpdateSuccess)
+      // However, this time, don't trigger a pending refresh state.
+      tryApplyUpdates(
+        hasUpdates ? undefined : onBeforeHotUpdate,
+        hasUpdates ? onBuildOk : onHotUpdateSuccess
+      )
```

Comment:
Are we sure we don't want to trigger another pending refresh state if a new update came in while we were updating? A new update might clear up the errors that happened _during_ the previous update. Furthermore, in the exit condition, `onBuildOk` will be called **twice** instead of just once, since we will also call in the `else` branch below. I believe this logic is incorrect and we should do the following instead:

```suggestion
      tryApplyUpdates(
        onBeforeHotUpdate,
        onHotUpdateSuccess
      )
```

In which case we no longer need to pass these two functions as arguments to `tryApplyUpdates`, we can inline them directly instead.

Reply:
That makes sense to me.